### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by Rifat8080**
In this branch, I have:

Created a class Classroom with the following:
The @label instance variable is initialized in the constructor.
Setter and getter for @label (remember about attr_accessor).
Created the has-many/belongs to the relationship between ClassroomandStudent`. The following are implemented:
Created the has-many side (a classroom has many students).
Created the belongs-to side (a student belongs to a classroom).
When adding a student to a classroom it also sets the classroom for the student.
When setting the classroom for a student it also adds it to the classroom's students.
Created a class Book with the following:
@title and @author instance variables, are initialized in the constructor.
Setters and getters for instance variables (remember about attr_accessor).
Created a class Rental with the following:
The @date instance variable, is initialized in the constructor.
Setter and getter for @date (remember about attr_accessor).
Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following are implemented:
Created the has-many side of Book and Rental (a book has many rentals).
Created the belongs-to side of Rental and Book (a rental belongs to a book).
Created the has-many side of Person and Rental (a person has many rentals).
Created the belongs-to side of Rental and Person (a rental belongs to a person).
Modified the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Rental
+  attr_accessor :date
+  attr_reader :person, :book
```
Comment (Efiamotu-1) [inline rental.rb]: - [ ] You are required to create `getters & setters` for the `person & book` so using `attr_accessor` would be appropriate in this instance, Please kindly add this to your code and resubmit for approval :+1: :sparkles:

File: person.rb [hunk 2]
```diff
@@ -1,17 +1,32 @@
-class Person
+require './nameable'
+require './capitilize_decorator'
+require './trimmer_decorator'
+
+class Person < Nameable
+  attr_accessor :name, :age
+  attr_reader :id, :rentals
```
Comment (Efiamotu-1) [inline person.rb]: - [ ] You are required to create `getters & setters` for the `:name, :age, :rentals` so using `attr_accessor` would be appropriate in this instance for the `rentals`, Please kindly add this to your code and resubmit for approval :+1: :sparkles:

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @Rifat8080 , 

 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zeff96 — APPROVED** [review]
### Status: APPROVED :green_circle: 
Hi @Rifat8080 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

congrats

## Highlights
:heavy_check_mark: Linters passing
:heavy_check_mark: Correct GitFlow

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**